src/routing/approved.rs

```diff
@@ -227,25 +234,29 @@ impl Approved {
                     let addrs = section.peers().map(Peer::addr).copied().collect();
                     GetSectionResponse::Redirect(addrs)
                 };
-                let response = Query::GetSectionResponse(response);
+                let response = InfrastructureMessage::GetSectionResponse(response);
                 debug!("Sending {:?} to {}", response, sender);
 
                 vec![Command::SendMessage {
                     recipients: vec![sender],
                     delivery_group_size: 1,
-                    message: MessageType::InfrastructureQuery(response),
+                    message: MessageType::InfrastructureMessage(response),
                 }]
             }
-            Query::GetSectionResponse(_) => {
+            InfrastructureMessage::GetSectionResponse(_) => {
                 if let Some(RelocateState::InProgress(tx)) = &mut self.relocate_state {
                     trace!("Forwarding {:?} to the bootstrap task", message);
                     let _ = tx
-                        .send((MessageType::InfrastructureQuery(message), sender))
+                        .send((MessageType::InfrastructureMessage(message), sender))
                         .await;
                 }
 
                 vec![]
             }
+            InfrastructureMessage::InfrastructureError(_) => {
+                // TODO handle this...
```

Comment:
Not sure what to be doing there

Reply:
InfrastructureError shall only be handled by a client?

Reply:
I think we are trying to avoid using `unimplemented` within the production code.
you can log it as an error instead ?

Reply:
I can, but we need ot handle it no? I don't know what should we be doing there? Or you're thinking to handle it once the PR is in?

Reply:
> InfrastructureError shall only be handled by a client?
It could be handled at nodes, or routing even I guess? Anywhere we could receive that